ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/xpn/byo-hosted-zone/provision/cucushift-installer-rehearse-gcp-ipi-xpn-byo-hosted-zone-provision-chain.yaml

```diff
@@ -1,13 +1,14 @@
 chain:
-  as: cucushift-installer-rehearse-gcp-ipi-xpn-minimal-permission-byo-hosted-zone-provision
+  as: cucushift-installer-rehearse-gcp-ipi-xpn-byo-hosted-zone-provision
   steps: 
     - chain: ipi-conf-gcp
     - ref: ipi-conf-credentials-mode
     - ref: ipi-conf-gcp-xpn-network
     - ref: ipi-conf-gcp-create-firewall-rules
-    - ref: gcp-provision-private-dns-zone-in-service-project
-    - ref: gcp-provision-xpn-byo-hosted-zone-min-perm-passthrough-serviceaccount
+    - ref: gcp-provision-private-dns-zone
+    - ref: gcp-provision-xpn-min-perm-passthrough-serviceaccount
```

Comment:
The difference between the sa in `gcp-provision-xpn-min-perm-passthrough-serviceaccount` and `gcp-provision-xpn-byo-hosted-zone-min-perm-passthrough-serviceaccount` is dns permission, right ? Shall we merge the two steps to generate different sa per `PRE_CREATE_PRIVATE_ZONE` flag ?

Reply:
FYI The key differences between the 2 service accounts are, the first SA has permission `dns.networks.bindPrivateDNSZone` in the host project, and role DNS Administrator in the third project; the second SA has permissions `compute.firewalls.delete` & `compute.networks.updatePolicy` in the host project.

Reply:
So `gcp-provision-xpn-byo-hosted-zone-min-perm-passthrough-serviceaccount` is still required as a test point as before ?

Reply:
Updated `gcp-provision-xpn-min-perm-passthrough-serviceaccount`, to use the second SA if `PRIVATE_ZONE_PROJECT` is empty (acting the similar way as `ipi-conf-gcp-private-dns-zone`. WDYT?

Reply:
Okay, saw you already merged `gcp-provision-xpn-min-perm-passthrough-serviceaccount` and `gcp-provision-xpn-byo-hosted-zone-min-perm-passthrough-serviceaccount` by judging `PRIVATE_ZONE_PROJECT`.